src/domain/settings/return-reasons/detail.tsx

```diff
@@ -0,0 +1,123 @@
+import { navigate } from "gatsby"
+import {
+  useAdminDeleteReturnReason,
+  useAdminUpdateReturnReason,
+} from "medusa-react"
+import React, { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import DuplicateIcon from "../../../components/fundamentals/icons/duplicate-icon"
+import TrashIcon from "../../../components/fundamentals/icons/trash-icon"
+import Input from "../../../components/molecules/input"
+import BodyCard from "../../../components/organisms/body-card"
+import DeletePrompt from "../../../components/organisms/delete-prompt"
+import useModal from "../../../hooks/use-modal"
+import useToaster from "../../../hooks/use-toaster"
+import { getErrorMessage } from "../../../utils/error-messages"
+import CreateReturnReasonModal from "./create-reason-modal"
+
+const ReturnReasonDetail = ({ reason }) => {
+  const {
+    isOpen: showDuplicateModal,
+    handleOpen: handleOpenDuplicateModal,
+    handleClose: handleCloseDuplicateModal,
+  } = useModal()
+  const {
+    isOpen: showDanger,
+    handleClose: handleClosePrompt,
+    handleOpen: handleOpenPrompt,
+  } = useModal()
+  const { register, reset, handleSubmit } = useForm()
+  const toaster = useToaster()
+  const deleteRR = useAdminDeleteReturnReason(reason?.id)
+  const updateRR = useAdminUpdateReturnReason(reason?.id)
+
+  const handleDeletion = async () => {
+    deleteRR.mutate(null)
+  }
```

Comment:
nit: Reset `selectedReason` in parent component or refresh data, since right now, if you delete the currently selected reason, it stays in the dom.

Reply:
good catch @olivermrbl! Deletion is triggered in the child component so two ways we can approach this I think:

1. Create a callback function `onDelete` in the parent and pass it down to the `ReturnReasonDetail` component so it can fire it after the deletion takes place
2. Listen to when the return reasons are refetched, then change the selected return reason from the parent. As we've discussed in the past, the refetch happens automatically after a deletion.

Approach 1. is totally valid, but I'm not the biggest fan because in my opinion it adds a little bit of noise if you're reading the `ReturnReasonDetail` component. You'll find an `onDelete()` function being called, and you'll have to go back to see what that function does, then you'll have two components that are changing state (`selectedReason`) instead of one. I personally much rather have whatever `onDelete()` wants to do, to just do it in the parent since it's the one creating and managing the `selectedReason` state without the need to pass a callback down its child.

Took advantage of those react-query `onSuccess` callbacks to do the sorting, as well as figure out which element to select when a query is first fetched + refetched. Just wanted to try it out at first, but then noticed that it reads nicely compared to approach 1 and is much cleaner (no `useEffect`s needed).

Might be overthinking a little here :sweat_smile: but thought I'd share nevertheless.

Reply:
Very neat! Def. think this is the better approach @zakariaelas - nicely done! 👍